logic/broadcaster.js

```diff
@@ -199,15 +199,11 @@ Broadcaster.prototype.broadcast = function(params, options, cb) {
 					peers = peers.slice(0, self.config.broadcastLimit);
 				}
 
-				async.eachLimit(
-					peers,
-					self.config.parallelLimit,
-					peer => peer.rpc[options.api](options.data),
-					() => {
-						library.logger.debug('End broadcast');
-						return setImmediate(waterCb, null, peers);
-					}
-				);
+				peers
+					.slice(0, self.config.peerLimit)
+					.map(peer => peer.rpc[options.api](options.data));
```

Comment:
It might be nice to separate RPCs from events at some point in the future because now they both get added to the same `peer.rpc` object which can be confusing. Can be done later though.